tools/src/dbgen/Lib.hs

```diff
@@ -257,18 +257,17 @@ generateFakeTxs SimpleTxsHistory{..} aId   = do
         walletId = aiWId aId
 
     -- Generate arbitrary tx ids and txs for the fake history.
-    fakeTxIds <- liftIO $ replicateM txsNumber genTxId
     fakeTxs   <- liftIO $ replicateM txsNumber genTxHistoryEntry
 
     let fakeMapTxs :: Map TxId TxHistoryEntry
-        fakeMapTxs = fromList $ zip fakeTxIds fakeTxs
+        fakeMapTxs = fromList $ zip (map getTxId fakeTxs) fakeTxs
 
     -- Insert into the @WalletStorage@.
     insertIntoHistoryCache db walletId fakeMapTxs
   where
 
-    genTxId :: IO TxId
-    genTxId = generate arbitrary
+    getTxId :: TxHistoryEntry -> TxId
+    getTxId (THEntry txId _ _ _ _ _) = txId
```

Comment:
If one day new field appears in `TxHistoryEntry`, this line will be the only one which breaks 😒

Reply:
You are right. Fixed.